### ISSUE TITLE
fix lookup of Commands

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#1531]
```m2
i1 : lookup(help, String)

i2 : lookup(help#0, String)

o2 = -*Function[../../Macaulay2/m2/help.m2:379:20-384:28]*-

o2 : FunctionClosure
```

Perhaps we should make the lookup defined in hashtables.dd unprotected and redefine it in Core to check whether the first element in the sequence is a Command, and if so replace it with the function enclosed in the Command.

**Comment by mahrud** [comment]
This would also fix all of the following:
```m2
i1 : code (help, String)

o1 = -- no method function found: help(String)

i2 : help (help, String)
stdio:2:1:(3): error: expected (help,String) to be a method

i3 : locate (help, String)

i4 : debug Core

i5 : html TO (help, String)

o5 = <tt>help(String)</tt> (missing documentation<!-- tag: (help,String) -->)

```

**Comment by DanGrayson** [comment]
I think of "lookup" as a low-level quick function, and wouldn't like to slow it down. But making `help (help, String)` work seems sensible.

**Comment by mahrud** [comment]
I think the solution that I offered, where lookup is low level up to a certain point, then becomes a method function, would keep it reasonably fast before the point where it is redefined, so it should be okay.

**Comment by DanGrayson** [comment]
No, because getting the value of a variable such as "lookup" happens at run time, not at compile time.

**Comment by mahrud** [comment]
Fine, use lookup' function internally and for help, code, locate, etc. use the lookup method. Not that complicated.

**Comment by DanGrayson** [comment]
That would work.
